python/ray/data/_internal/dataset_logger.py

```diff
@@ -0,0 +1,44 @@
+import logging
+import os
+from typing import Optional
+
+import ray
+from ray._private.ray_constants import LOGGER_FORMAT
+from ray.util.annotations import DeveloperAPI
+
+
+@DeveloperAPI
```

Comment:
let's make it internal first, we can add `DeveloperAPI` later if it's needed by others.